IntelliTect.TestTools.TestFramework/IntelliTect.TestTools.TestFramework/DebugLogger.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Text.Json;
+
+namespace IntelliTect.TestTools.TestFramework
+{
+    public class DebugLogger : ITestCaseLogger
+    {
+        public DebugLogger(TestCase testCase)
+        {
+            TestCase = testCase;
+        }
+
+        public TestCase TestCase { get; }
+        public string? CurrentTestBlock { get; set; }
+
+        public void Debug(string message)
+        {
+            LogToDebug($"{TestCase.TestMethodName} - {CurrentTestBlock} - Debug: {message}");
+        }
+
+        public void Critical(string message)
+        {
+            LogToDebug($"{TestCase.TestMethodName} - {CurrentTestBlock} - Error: {message}");
+        }
+
+        public void Info(string message)
+        {
+            LogToDebug($"{TestCase.TestMethodName} - {CurrentTestBlock} - Info: {message}");
+        }
+
+        public void TestBlockInput(object input)
+        {
+            string inputString = Serialize(input);
+            LogToDebug($"{TestCase.TestMethodName} - {CurrentTestBlock} - Input arguments: {inputString}");
+        }
+
+        public void TestBlockOutput(object output)
+        {
+            string outputString = Serialize(output);
+            LogToDebug($"{TestCase.TestMethodName} - {CurrentTestBlock} - Output returns: {outputString}");
+        }
+
+        private void LogToDebug(object message)
+        {
+            System.Diagnostics.Debug.WriteLine(message);
+        }
+
+        private string Serialize(object objectToParse)
```

Comment:
I see a null check on this parameter in the catch block. Is it possible that a null gets in here?

Reply:
I'm not 100% sure how the nullable property propagates to consuming libraries. So... maybe? :) I think, based off of a prior comment you made, probably. If a person consumes this package without nullability enabled in their own project, they can invoke any of the public methods with their own code, so I'd be worried they could invoke with a null value. I don't want to just throw in those scenarios, but maybe a better message is needed in the catch block?

Reply:
Nullable reference work by the compiler adding attributes to your public API. Consumers with a compiler that also has the feature enabled then use these for the null analysis.

So for this, I would suggest either null checking the argument and throwing an `ArgumentNullException` if you get a null, or declare the parameter as nullable, and simply return empty string if the argument is null.